firmware_tools/ghidra/vxhunter_utility/symbol.py

```diff
@@ -422,58 +352,66 @@ def demangled_symbol(symbol_string):
                 sym_demangled = demangler.demangle(symbol_string, False)
 
         except DemangledException as err:
-            logger.debug("DemangledException: symbol_string: {}, reason:{}".format(symbol_string, err))
+            logger.debug("First pass demangling failed: symbol_string: {}, reason: {}".format(symbol_string, err))
+            pass
 
-        try:
-            if not sym_demangled:
+        if not sym_demangled:
+            try:
                 # Temp fix to handle _ prefix function name by remove _ prefix before demangle
                 sym_demangled = demangler.demangle(symbol_string[1:], False)
 
-        except DemangledException as err:
-            logger.debug("DemangledException: symbol_string: {}, reason:{}".format(symbol_string, err))
+            except DemangledException as err:
+                logger.debug("Second pass demangling failed: symbol_string: {}, reason:{}".format(symbol_string, err))
+                pass
 
         if sym_demangled:
             sym_demangled_name = sym_demangled.getSignature(False)
 
-        if sym_demangled_name:
-            logger.debug("sym_demangled_name: {}".format(sym_demangled_name))
+            if sym_demangled_name:
+                logger.debug("sym_demangled_name: {}".format(sym_demangled_name))
+            else:
+                logger.debug("Demangled symbol name for string {} is None.".format(symbol_string))
 
     return sym_demangled_name
 
 
 def add_symbol(symbol_name, symbol_name_address, symbol_address, symbol_type):
     symbol_address = toAddr(symbol_address)
     symbol_name_string = symbol_name
-
     # Get symbol_name
     if symbol_name_address:
         symbol_name_address = toAddr(symbol_name_address)
-        if getDataAt(symbol_name_address):
-            logger.debug("removeDataAt: {}".format(symbol_name_address))
-            removeDataAt(symbol_name_address)
-
+        logger.debug("Have symbol name {} at address {}.".format(symbol_name_string, symbol_name_address))
+        # There might be a problem here. It seems like we're getting exceptions because bogus data exists at locations that
+        # never gets removed because this code doesn't run...
         try:
             symbol_name_string = createAsciiString(symbol_name_address).getValue()
-            logger.debug("symbol_name_string: {}".format(symbol_name_string))
-
+            logger.debug("Created ascii string {} at {}.".format(symbol_name_string, symbol_name_address))
         except CodeUnitInsertionException as err:
-            logger.error("Got CodeUnitInsertionException: {}".format(err))
-
-        except:
+            logger.error("Failed to create ascii string for symbol named {} at {}: {}".format(symbol_name, symbol_name_address, err))
+        except Exception as err:
+            logger.error("Failed to create ascii string for symbol named {} at {}: {}; returning.".format(symbol_name, symbol_name_address, err))
             return
```

Comment:
Please add `except:` at the end, sometimes ghidra will raise exceptions which can't be captured using `except Exception`.

Reply:
Instead of doing `except Exception as...`, I've changed it to do `except BaseException as...`. This is functionally equivalent to a bare `except...`, with the added benefit that we get to log the message that comes out -- what ever that might be. I hope that you enjoy this change!